static/src/javascripts/projects/common/modules/commercial/adblock-messages.js

```diff
@@ -32,18 +34,20 @@ define([
     }
 
     function noAdblockMsg() {
-        return adblockInUseSync() && notMobile() && (
+        return adblockInUse().then(function(adblockInUse){
```

Comment:
We could improve this expression and the next, by delaying the adblock test as much as possible:

```
return notMobile() && (!visitedMoreThanOnce() || !isAdblockSwitchOn() || (visitedMoreThanOnce() && isPayingMember())) ?
  adblockInUse() :
  Promise.resolve(false);
```

(notice I removed the redundant `isAdblockSwitchOn` call in the third OR clause)

I'll leave you the one in `showAdblockMsg` as a small exercise.